code/modules/manufactorio/machines/unloader.dm

```diff
@@ -0,0 +1,75 @@
+/obj/machinery/power/manufacturing/unloader
+	name = "manufacturing crate unloader"
+	desc = "Unloads crates (and ore boxes) passed into it, ejecting the empty crate to the side and its contents forwards. Use a multitool to flip the crate output."
+	icon = 'icons/obj/machines/mining_machines.dmi'
+	icon_state = "unloader-corner"
+	circuit = /obj/item/circuitboard/machine/manuunloader
+	/// power used per attempt to unload a crate
+	var/power_to_unload_crate = 2 KILO WATTS
+	/// whether the side we output unloaded crates is flipped
+	var/flip_side = FALSE
+
+/obj/machinery/power/manufacturing/unloader/update_overlays()
+	. = ..()
+	. += generate_io_overlays(dir, COLOR_ORANGE) // OUT - stuff in it
+	. += generate_io_overlays(REVERSE_DIR(dir), COLOR_MODERATE_BLUE) // IN - crate
+	. += generate_io_overlays(turn(dir, flip_side ? 90 : -90), COLOR_ORANGE) // OUT -- empty crate
+
+/obj/machinery/power/manufacturing/unloader/request_resource() //returns held crate if someone wants to do that for some reason
+	if(!length(contents - circuit))
+		return
+	return (contents - circuit)[1]
+
+/obj/machinery/power/manufacturing/unloader/multitool_act(mob/living/user, obj/item/tool)
+	. = ..()
+	balloon_alert(user, "flipped")
+	flip_side = !flip_side
+	update_appearance()
+
+/obj/machinery/power/manufacturing/unloader/receive_resource(obj/receiving, atom/from, receive_dir)
+	if(surplus() < power_to_unload_crate || receive_dir != REVERSE_DIR(dir))
+		return MANUFACTURING_FAIL
+	if(length(contents - circuit))
+		return MANUFACTURING_FAIL_FULL
+
+	var/obj/structure/closet/as_closet = receiving
+	var/obj/structure/ore_box/as_orebox = receiving
+	if(istype(as_closet))
+		if(!as_closet.can_open())
+			return MANUFACTURING_FAIL
+	else if(!istype(as_orebox))
+		return MANUFACTURING_FAIL
+	receiving.Move(src, get_dir(receiving, src))
+	START_PROCESSING(SSfastprocess, src)
+	return MANUFACTURING_SUCCESS
+
+/obj/machinery/power/manufacturing/unloader/process(seconds_per_tick)
+	if(!length(contents - circuit))
```

Comment:
yeah, define a list var.

Reply:
A list var cant be done without modifying it again later everywhere; most machines cause contents to be modified during processing and usually check it after that again